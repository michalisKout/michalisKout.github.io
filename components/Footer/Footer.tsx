import FooterTitle from "./components/FooterTitle";
import Subscribe from "./components/Subscribe";
import PrimaryLink from "../Shared/PrimaryLink";
import NavItems from "../Shared/NavItems";

const Footer = () => (
  <footer className="footer">
    <div className="container py-6">
      <FooterTitle />
      <div className="flex justify-center mt-6">
        <Subscribe />
      </div>

      <hr className="breaking-line" />

      <div className="footer--menu">
        <PrimaryLink />

        <div className="footer--menu-wrapper">
          <NavItems
            wrapperClassName={"-mx-4"}
            itemClassName={"footer--menu--link"}
          />
        </div>
      </div>
    </div>
  </footer>
);

export default Footer;
