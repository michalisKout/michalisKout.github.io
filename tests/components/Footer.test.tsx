import { render } from "@testing-library/react";
import Footer from "../../components/Footer/Footer";

it("should render Footer", () => {
  const { asFragment } = render(<Footer />);
  expect(asFragment()).toMatchSnapshot();
});
