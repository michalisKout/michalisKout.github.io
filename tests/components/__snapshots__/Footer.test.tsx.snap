// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container py-6"
    >
      <h1
        class="footer--title"
      >
        Join 31,000+ other and never miss 
        <br />
         out on new tips, tutorials, and more.
      </h1>
      <div
        class="flex justify-center mt-6"
      >
        <div
          class="subscribe--container"
        >
          <div
            class="subscribe--wrapper"
          >
            <input
              aria-label="Enter your email"
              class="subscribe--input"
              placeholder="Enter your email"
              type="email"
            />
            <button
              class="button-primary"
            >
              subscribe
            </button>
          </div>
        </div>
      </div>
      <hr
        class="breaking-line"
      />
      <div
        class="footer--menu"
      >
        <div>
          <a
            class="link-primary"
            href="#"
          >
            Brand
          </a>
        </div>
        <div
          class="footer--menu-wrapper"
        >
          <div
            class="-mx-4"
          >
            <span
              class="footer--menu--link"
              data-testid="nav-item"
            >
              <a
                href="/"
              >
                Home
              </a>
            </span>
            <span
              class="footer--menu--link"
              data-testid="nav-item"
            >
              <a
                href="/portfolio"
              >
                Portfolio
              </a>
            </span>
            <span
              class="footer--menu--link"
              data-testid="nav-item"
            >
              <a
                href="/experience"
              >
                Experience
              </a>
            </span>
            <span
              class="footer--menu--link"
              data-testid="nav-item"
            >
              <a
                href="/contact"
              >
                Contact
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
